### PR TITLE
:art: Upgrade ruff and enable pyupgrade checks

### PR DESCRIPTION
**Pull request by chassing**
Upgrade to the latest `ruff` version and enable `pyupgrade` checks. `pyupgrade` takes care of obsolete things, like deprecated imports or new Python features like `foobar : str | None ...` instead of `foobar: Optional[str] ...`

Please review commit by commit!

### CODE REVIEW

File: release/version.py [hunk 1]
```diff
@@ -47,7 +48,7 @@ def commit(length: int = 7) -> str:
     return p.stdout.decode("utf-8").strip()
 
 
-def semver(git_version: str | None = None) -> str:
+def semver(git_version: Optional[str] = None) -> str:  # noqa: UP007 - RHEL8 has python 3.8
```
Comment (hemslo) [inline release/version.py:51]: we should create a card to make release run in container so no dependencies on ci host

**Review by hemslo — APPROVED** [review]
Nice upgrade and cleanup!

File: release/version.py (hunk 1 above)
Comment (chassing) [reply release/version.py:51]: Done https://issues.redhat.com/browse/APPSRE-10403

**Comment by patjlm** [comment]
❤️

**Pull request merged**